applications/pelion_client/src/modules/pelion_fcc.cpp

```diff
@@ -5,42 +5,37 @@
  */
 
 #include <zephyr.h>
-
+#include "factory_reset_event.h"
 #include "factory_configurator_client.h"
 #include "pelion_fcc_err.h"
 
-#include <caf/events/button_event.h>
 
 #define MODULE pelion_fcc
 #include <caf/events/module_state_event.h>
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_PELION_CLIENT_PELION_LOG_LEVEL);
 
+#define NET_MODULE_ID_STATE_READY_WAIT_FOR                               \
+	(IS_ENABLED(CONFIG_PELION_CLIENT_FACTORY_RESET_REQUEST_ENABLE) ? \
+		MODULE_ID(factory_reset_request) : MODULE_ID(main))
 
-static struct k_delayed_work fcc_init_work;
-static bool reset_storage;
+static bool initialized;
 
 
-static void fcc_init_work_handler(struct k_work *work)
+static void handle_init(void)
 {
-	fcc_status_e err = fcc_init();
+	fcc_status_e err;
+
+	__ASSERT_NO_MSG(!initialized);
+
+	err = fcc_init();
```

Comment:
```
fcc_status_e err = fcc_init();
```

Reply:
The reason I split variable declaration from its use is the fact that checkpatch requires an empty line after such usage like above. It looks odd when below we set err without variable declaration.

```
fcc_status_e err = init();

if (err) {
...
}

err = do_something_else();
if (err) {
...
}
```

In my option:

```
fcc_status_e err;

err = init();
if (err) {
...
}

err = do_something_else();
if (err) {
...
}
```